skore/src/skore/ui/project_routes.py

```diff
@@ -1,97 +1,135 @@
-                value = item.estimator_html_repr
-                media_type = "application/vnd.sklearn.estimator+html"
-            elif isinstance(item, MediaItem):
-                value = base64.b64encode(item.media_bytes).decode()
-                media_type = item.media_type
-            elif isinstance(item, CrossValidationItem):
-                value = base64.b64encode(item.plot_bytes).decode()
-                media_type = "application/vnd.plotly.v1+json"
-            else:
-                raise ValueError(f"Item {item} is not a known item type.")
-
-            items[key].append(
-                SerializedItem(
-                    media_type=media_type,
-                    value=value,
-                    updated_at=item.updated_at,
-                    created_at=item.created_at,
-                )
-            )
-
-    views = {key: project.get_view(key).layout for key in project.list_view_keys()}
-
-    return SerializedProject(
-        items=dict(items),
-        views=views,
-    )
+class JSONError(Exception):
+    """Exception for objects that can't be serialized."""
+
+
+def default(_):
+    """Raise `JSONError` for objects that can't be serialized."""
+    raise JSONError
+
+
+class ProjectResponse(JSONResponse):
+    """Project response with our own serialization strategy."""
+
+    @staticmethod
+    def serialize_item_to_json(item: Item) -> str:
+        """Serialize Item to JSON."""
+        if isinstance(item, PrimitiveItem):
+            value = item.primitive
+            media_type = "text/markdown"
+        elif isinstance(item, NumpyArrayItem):
+            value = item.array.tolist()
+            media_type = "text/markdown"
+        elif isinstance(item, PandasDataFrameItem):
+            value = item.dataframe.to_dict(orient="tight")
+            media_type = "application/vnd.dataframe+json"
+        elif isinstance(item, PandasSeriesItem):
+            value = item.series.to_list()
+            media_type = "text/markdown"
+        elif isinstance(item, SklearnBaseEstimatorItem):
+            value = item.estimator_html_repr
+            media_type = "application/vnd.sklearn.estimator+html"
+        elif isinstance(item, MediaItem):
+            value = base64.b64encode(item.media_bytes).decode()
+            media_type = item.media_type
+        elif isinstance(item, CrossValidationItem):
+            value = base64.b64encode(item.plot_bytes).decode()
+            media_type = "application/vnd.plotly.v1+json"
+        else:
+            raise ValueError(f"Item {item} is not a known item type.")
+
+        try:
+            value = json.dumps(value, default=default)
+        except JSONError:
+            value = json.dumps(None)
+            error = True
+            traceback = format_exc()
+        else:
+            error = False
+            traceback = None
+
+        media_type = json.dumps(media_type)
+        error = json.dumps(error)
+        traceback = json.dumps(traceback)
```

Comment:
Do you use json encoding to escape double quotes from the traceback ?
As the traceback wont be easily displayable in the frontend, could we have something more concise to present to the user ?